packages/cli/src/commands/generate/schema-from-database/generate_schema_command.ts

```diff
@@ -0,0 +1,124 @@
+import { Argv, CommandModule } from 'yargs';
+import { BackendIdentifierResolver } from '../../../backend-identifier/backend_identifier_resolver.js';
+import { ArgumentsKebabCase } from '../../../kebab_case.js';
+import { SecretClient } from '@aws-amplify/backend-secret';
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import { SchemaGenerator } from '@aws-amplify/schema-generator';
+
+const DEFAULT_OUTPUT = 'schema.sql.ts';
+
+export type GenerateSchemaCommandOptions =
+  ArgumentsKebabCase<GenerateSchemaCommandOptionsCamelCase>;
+
+type GenerateSchemaCommandOptionsCamelCase = {
+  stack: string | undefined;
+  appId: string | undefined;
+  branch: string | undefined;
+  out: string | undefined;
+  connectionStringSecret: string | undefined;
+};
+
+/**
+ * Command that generates typescript data schema from sql schema.
+ */
+export class GenerateSchemaCommand
+  implements CommandModule<object, GenerateSchemaCommandOptions>
+{
+  /**
+   * @inheritDoc
+   */
+  readonly command: string;
+
+  /**
+   * @inheritDoc
+   */
+  readonly describe: string;
+
+  /**
+   * Creates typescript data schema generation command.
+   */
+  constructor(
+    private readonly backendIdentifierResolver: BackendIdentifierResolver,
+    private readonly secretClient: SecretClient,
+    private readonly schemaGenerator: SchemaGenerator
+  ) {
+    this.command = 'schema-from-database';
+    this.describe = 'Generates typescript data schema from a SQL database';
+  }
+
+  getBackendIdentifier = async (args: GenerateSchemaCommandOptions) => {
+    return await this.backendIdentifierResolver.resolve(args);
+  };
+
+  /**
+   * @inheritDoc
+   */
+  handler = async (args: GenerateSchemaCommandOptions): Promise<void> => {
+    const backendIdentifier = await this.backendIdentifierResolver.resolve(
+      args
+    );
+
+    if (!backendIdentifier) {
+      throw new Error('Could not resolve the backend identifier');
```

Comment:
Assuming that not specifying correct args to the command never gets to this point, this should be an `AmplifyFault`.

Reply:
Updated.